gnome-base/gnome-shell/files/44.1-Disable-HAVE_EXE_INTROSPECTION.patch

```diff
@@ -0,0 +1,28 @@
+From a7a4882de3747ee4ff9ef53272e1063b3394fc0d Mon Sep 17 00:00:00 2001
+From: Mike Rostecki <vadorovsky@protonmail.com>
+Date: Mon, 12 Jun 2023 16:57:04 +0200
+Subject: [PATCH] Disable HAVE_EXE_INTROSPECTION
+
+The EXE introspection feature does not work on musl and it relies
+on the _DYNAMIC variable in link.h, which exists only in glibc.
+---
+ meson.build | 3 +--
+ 1 file changed, 1 insertion(+), 2 deletions(-)
+
+diff --git a/meson.build b/meson.build
+index 7b08fe52d..5ec06d371 100644
+--- a/meson.build
++++ b/meson.build
+@@ -148,8 +148,7 @@ cdata.set('HAVE_FDWALK', cc.has_function('fdwalk'))
+ cdata.set('HAVE_MALLINFO', cc.has_function('mallinfo'))
+ cdata.set('HAVE_MALLINFO2', cc.has_function('mallinfo2'))
+ cdata.set('HAVE_SYS_RESOURCE_H', cc.has_header('sys/resource.h'))
+-cdata.set('HAVE_EXE_INTROSPECTION',
```

Comment:
Can you send this upstream with a patch for it to check for `_DYNAMIC` properly (or guard it entirely on glibc if it's using internals)?

Reply:
Hm, see https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2747. They seem to think it's fine on musl even.

Reply:
Sure, I will try to fix that upstream.

Reply:
I'm not sure why you closed this, I was asking for two things:
1. How this squares with https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2747 which implies this shouldn't be a problem
1. For you to send it upstream and include a link in the patch

Reply:
Sorry for misunderstanding.

https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2747 implies that executable introspection works on Linux with musl, but that isn't true (at least from my experience). On my Gentoo with musl+LLVM profile, building gnome-shell fails with:

```
[266/280] Compiling C object src/gnome-shell.p/main.c.o
FAILED: src/gnome-shell.p/main.c.o
cc -Isrc/gnome-shell.p -Isrc -I../src -I. -I.. -Isrc/st -I../src/st -Isrc/tray -I../src/tray -I/usr/include/gio-unix-2.0 -I/usr/include/glib-2.0 -I/usr/lib/glib-2.0/include -I/usr/lib/libffi/include -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/libxml2 -I/usr/include/at-spi2-atk/2.0 -I/usr/include/at-spi-2.0 -I/usr/include/dbus-1.0 -I/usr/lib/dbus-1.0/include -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/gjs-1.0 -I/usr/include/gobject-introspection-1.0 -I/usr/include/nspr -I/usr/include/cairo -I/usr/include/freetype2 -I/usr/include/pixman-1 -I/usr/include/mutter-12/clutter -I/usr/include/json-glib-1.0 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/fribidi -I/usr/include/mutter-12/cogl -I/usr/include/graphene-1.0 -I/usr/lib/graphene-1.0/include -I/usr/include/startup-notification-1.0 -I/usr/include/polkit-1 -I/usr/include/gcr-4 -I/usr/include/p11-kit-1 -I/usr/include/gck-2 -I/usr/include/libnm -I/usr/include/libsecret-1 -I/usr/include/mutter-12 -I/usr/include/gsettings-desktop-schemas -fdiagnostics-color=always -D_FILE_OFFSET_BITS=64 -Wall -Winvalid-pch -O0 -g -fno-strict-aliasing -Wpointer-arith -Wmissing-declarations -Wimplicit-function-declaration -Wformat=2 -Wformat-nonliteral -Wformat-security -Wstrict-prototypes -Wmissing-prototypes -Wnested-externs -Wold-style-definition -Wundef -Wunused -Wcast-align -Wmissing-noreturn -Wmissing-format-attribute -Wmissing-include-dirs -Wlogical-op -Wignored-qualifiers -Werror=redundant-decls -Werror=implicit -Werror=nonnull -Werror=init-self -Werror=main -Werror=missing-braces -Werror=sequence-point -Werror=return-type -Werror=trigraphs -Werror=array-bounds -Werror=write-strings -Werror=address -Werror=int-to-pointer-cast -Werror=pointer-to-int-cast -Werror=empty-body -Werror=write-strings -DG_ENABLE_DEBUG -fno-omit-frame-pointer -pthread -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 -mfpmath=sse -msse -msse2 -isystem /usr/include/mozjs-102 -DCLUTTER_ENABLE_EXPERIMENTAL_API -DCOGL_ENABLE_EXPERIMENTAL_API '-DVERSION="44.3"' '-DLOCALEDIR="/usr/local/share/locale"' '-DDATADIR="/usr/local/share"' '-DGNOME_SHELL_LIBEXECDIR="/usr/local/libexec"' '-DGNOME_SHELL_DATADIR="/usr/local/share/gnome-shell"' '-DGNOME_SHELL_PKGLIBDIR="/usr/local/lib/gnome-shell"' '-DMUTTER_TYPELIB_DIR="/usr/lib/mutter-12"' -MD -MQ src/gnome-shell.p/main.c.o -MF src/gnome-shell.p/main.c.o.d -o src/gnome-shell.p/main.c.o -c ../src/main.c
../src/main.c: In function 'maybe_add_rpath_introspection_paths':
../src/main.c:143:14: error: '_DYNAMIC' undeclared (first use in this function); did you mean 'PT_DYNAMIC'?
  143 |   for (dyn = _DYNAMIC; dyn->d_tag != DT_NULL; dyn++)
      |              ^~~~~~~~
      |              PT_DYNAMIC
../src/main.c:143:14: note: each undeclared identifier is reported only once for each function it appears in
[273/280] Generating src/st/St-12.gir with a custom command (wrapped by meson to set env)
ninja: build stopped: subcommand failed.
```

And I took your comment as "please prepare a patch to gnome-shell, which guards the `_DYNAMIC` part on glibc, include a musl-compliant alternative for the guarded code, send it upstream, then come back here with the same patch". Which made sense for me, so I closed this PR, which just disables the whole feature entirely (like the upstream PR 2747 does for non-Linux systems). And unless you have any other suggestions, that's what I'm planning to do.

Also, sorry for late reaction, I will do my best to be more responsive and work on the proper fix now.